library/src/main/java/com/liulishuo/filedownloader/FileDownloadServiceUIGuard.java

```diff
@@ -248,6 +248,8 @@ public void stopForeground(boolean removeNotification) {
             getService().stopForeground(removeNotification);
         } catch (RemoteException e) {
             e.printStackTrace();
+        } finally {
+            runServiceForeground = false;
```

Comment:
You can't handle all cases from this, especially for the lost connection rather than stop manually, please set to `false` on the callback of `onServiceConnected`.

Reply:
Okay, because of the `stopForegroundService` is safe to invoke even if there isn't any service connected, but if we set to `false` on the `onServiceConnected` then we have to handle the service killed and re-created by system case, therefore for more simpleness you need to cover this case.